src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -640,13 +640,13 @@ impl<'tcx> Tree {
         // Register new_tag as a child of parent_tag
         self.nodes.get_mut(parent_idx).unwrap().children.push(idx);
 
-        // We need to know the biggest SIFA for `update_last_accessed_after_retag` below.
-        let mut max_sifa = default_strongest_idempotent;
+        // We need to know the weakest SIFA for `update_last_accessed_after_retag` below.
+        let mut min_sifa = default_strongest_idempotent;
         for (Range { start, end }, &perm) in
             inside_perms.iter(Size::from_bytes(0), inside_perms.size())
         {
             assert!(perm.permission.is_initial());
```

Comment:
can we add an assert here that `perm.idempotent_foreign_access == perm.permission.strongest_idempotent_foreign_access(protected)`? I do not fully recall what sets `perm.idempotent_foreign_access` initially and since it's "far away" from this function it might be easy to miss that this is used here.